pyHalo/realization_extensions.py

```diff
@@ -497,3 +502,201 @@ def _get_fluctuation_halos(realization, fluctuation_amplitude, fluctuation_size,
+       
+    corr = np.zeros((r.shape[0], mu.shape[0]))
+       
+    for i in range(r.shape[0]):
+        for j in range(mu.shape[0]):
+            x1 = xx_ - (r[i]/2)*((cos_phi*mu[j]) + (sin_phi*np.sqrt(1-mu[j]**2)))
+            y1 = yy_ - (r[i]/2)*((sin_phi*mu[j]) - (cos_phi*np.sqrt(1-mu[j]**2)))
+            
+            x2 = xx_ + (r[i]/2)*((cos_phi*mu[j]) + (sin_phi*np.sqrt(1-mu[j]**2)))
+            y2 = yy_ + (r[i]/2)*((sin_phi*mu[j]) - (cos_phi*np.sqrt(1-mu[j]**2)))
+            
+            if apply_mask == True:
+                if r_max == None:
+                    Area = (x_.max()-x_.min())*(y_.max()-y_.min()) - np.pi*(r_min**2) 
+                else:
+                    Area = np.pi*(r_max**2 - r_min**2)
+            else:
+                Area = (x_.max()-x_.min())*(y_.max()-y_.min())
+            
+            kappa_interp_1_ = kappa_interp(y1, x1, grid = False)
+            if apply_mask == True:
+                mask_interp_1 = mask_interp(y1, x1, grid = False)
+            else:
+                mask_interp_1 = np.ones(x1.shape)
+            
+            kappa_interp_2_ = kappa_interp(y2, x2, grid = False)
+            if apply_mask == True:
+                mask_interp_2 = mask_interp(y2, x2, grid = False)
+            else:
+                mask_interp_2 = np.ones(x2.shape)
+                
+            mask_interp_1[mask_interp_1<0.9] = 0
+            mask_interp_1[mask_interp_1>0.9] = 1
+
+            mask_interp_2[mask_interp_2<0.9] = 0
+            mask_interp_2[mask_interp_2>0.9] = 1 
+            
+            kappa_interp_1 = kappa_interp_1_*mask_interp_1
+            kappa_interp_2 = kappa_interp_2_*mask_interp_2
+            
+            term_1 = simps(simps(kappa_interp_1*kappa_interp_2, x_, axis=0), y_, axis=-1)
+            term_2_1 = simps(simps(kappa_interp_1*mask_interp_2, x_, axis=0), y_, axis=-1)
+            term_2_2 = simps(simps(mask_interp_1*kappa_interp_2, x_, axis=0), y_, axis=-1)
+            term_3 = simps(simps(mask_interp_1*mask_interp_2, x_, axis=0), y_, axis=-1)
+            
+            NCC_num = (term_1 - ((term_2_1*term_2_2)/term_3))/Area
+            
+            term_4 = simps(simps((kappa_interp_1**2)*mask_interp_2, x_, axis=0), y_, axis=-1)
+            term_5 = term_2_1**2
+            term_6 = term_3
+            
+            NCC_den_1 = (term_4 - (term_5/term_6))/Area
+            
+            term_7 = simps(simps(mask_interp_1*(kappa_interp_2**2), x_, axis=0), y_, axis=-1)
+            term_8 = term_2_2**2
+            term_9 = term_3
+            
+            NCC_den_2 = (term_7 - (term_8/term_9))/Area   
+            
+            if normalization == True:
+                corr[i,j] = NCC_num/np.sqrt(NCC_den_1*NCC_den_2) 
+            else:
+                corr[i,j] = NCC_num
+            
+    end_time = time.time()
+    
+    print(f"It took {end_time-start_time:.2f} seconds to compute the correlation map") 
+    
+    return corr
+
+def xi_l(l, corr, r, mu):
```

Comment:
since it is very unlikely anyone will use this function on its own (and probably nobody will want to import it into a notebook or any other file), we should make it a private python function. This will hide it from import * statements and makes the code easier to use. You do this simply by putting an underscore at the front of the function name, i.e. 

def _xi_l(l, corr, r, mu): 

instead of 

def xi_l(l, corr, r, mu):